openmc/mgxs/mgxs.py

```diff
@@ -1525,55 +1532,35 @@ def tallies(self):
     @property
     def rxn_rate_tally(self):
         if self._rxn_rate_tally is None:
-            self._rxn_rate_tally = self.tallies['fission']
+            self._rxn_rate_tally = self.tallies[self._rxn_type]
             self._rxn_rate_tally.sparse = self.sparse
         return self._rxn_rate_tally
 
 
-class NuFissionXS(MGXS):
-    """A fission production multi-group cross section."""
+class FissionXS(FissionXSBase):
+    """A fission multi-group cross section."""
 
     def __init__(self, domain=None, domain_type=None,
                  groups=None, by_nuclide=False, name=''):
-        super(NuFissionXS, self).__init__(domain, domain_type,
-                                          groups, by_nuclide, name)
-        self._rxn_type = 'nu-fission'
-
-    @property
-    def tallies(self):
-        """Construct the OpenMC tallies needed to compute this cross section.
-
-        This method constructs two tracklength tallies to compute the 'flux'
-        and 'nu-fission' reaction rates in the spatial domain and energy
-        groups of interest.
-
-        """
-
-        # Instantiate tallies if they do not exist
-        if self._tallies is None:
-
-            # Create a list of scores for each Tally to be created
-            scores = ['flux', 'nu-fission']
-            estimator = 'tracklength'
-            keys = scores
+        super(FissionXS, self).__init__('fission', domain, domain_type,
+                                        groups, by_nuclide, name)
 
-            # Create the non-domain specific Filters for the Tallies
-            group_edges = self.energy_groups.group_edges
-            energy_filter = openmc.Filter('energy', group_edges)
-            filters = [[energy_filter], [energy_filter]]
 
-            # Initialize the Tallies
-            self._create_tallies(scores, filters, keys, estimator)
+class NuFissionXS(FissionXSBase):
+    """A fission product multi-group cross section."""
 
-        return self._tallies
+    def __init__(self, domain=None, domain_type=None,
+                 groups=None, by_nuclide=False, name=''):
+        super(NuFissionXS, self).__init__('nu-fission', domain, domain_type,
+                                          groups, by_nuclide, name)
 
-    @property
-    def rxn_rate_tally(self):
-        if self._rxn_rate_tally is None:
-            self._rxn_rate_tally = self.tallies['nu-fission']
-            self._rxn_rate_tally.sparse = self.sparse
-        return self._rxn_rate_tally
+class KappaFissionXS(FissionXSBase):
+    """A fission product energy multi-group cross section."""
```

Comment:
This should be differentiated from nu-fission - perhaps "A recoverable fission energy production rate multi-group cross section"